src/ctapipe/calib/camera/extractor.py

```diff
@@ -0,0 +1,250 @@
+class SigmaClippingExtractor(StatisticsExtractor):
+    """
+    Extract the statistics from a chunk of charge or variance images
+    using astropy's sigma clipping functions
+    """
+
+    median_outliers_interval = List(
+        [-0.3, 0.3],
+        help=(
+            "Interval of the multiplicative factor for detecting outliers based on"
+            "the deviation of the median distribution."
+            "- If `outlier_method` is `median`, the factors are multiplied by"
+            "  the camera median of pixel medians to set the thresholds for identifying outliers."
+            "- If `outlier_method` is `standard_deviation`, the factors are multiplied by"
+            "  the camera standard deviation of pixel medians to set the thresholds for identifying outliers."
+        ),
+    ).tag(config=True)
+    outlier_method = CaselessStrEnum(
+        values=["median", "standard_deviation"],
+        help="Method used for detecting outliers based on the deviation of the median distribution",
+    ).tag(config=True)
+    std_outliers_interval = List(
+        [-3, 3],
+        help=(
+            "Interval of the multiplicative factor for detecting outliers based on"
+            "the deviation of the standard deviation distribution."
+            "The factors are multiplied by the camera standard deviation of pixel standard deviations"
+            "to set the thresholds for identifying outliers."
+        ),
+    ).tag(config=True)
+    max_sigma = Int(
+        default_value=4,
+        help="Maximal value for the sigma clipping outlier removal",
+    ).tag(config=True)
+    iterations = Int(
+        default_value=5,
+        help="Number of iterations for the sigma clipping outlier removal",
+    ).tag(config=True)
+
+    def extract(self, images, times, masked_pixels_of_sample) -> StatisticsContainer:
+        # Mask broken pixels
+        masked_images = np.ma.array(images, mask=masked_pixels_of_sample)
+
+        # Calculate the mean, median, and std over the chunk per pixel
+        pix_mean, pix_median, pix_std = sigma_clipped_stats(
+            masked_images,
+            sigma=self.max_sigma,
+            maxiters=self.iterations,
+            cenfunc="mean",
+            axis=0,
+        )
+
+        # Mask pixels without defined statistical values
+        pix_mean = np.ma.array(pix_mean, mask=np.isnan(pix_mean))
+        pix_median = np.ma.array(pix_median, mask=np.isnan(pix_median))
+        pix_std = np.ma.array(pix_std, mask=np.isnan(pix_std))
+
+        # Camera median of the pixel medians
+        cam_median_of_pix_median = np.ma.median(pix_median, axis=1)
+
+        # Camera median of the pixel stds
+        cam_median_of_pix_std = np.ma.median(pix_std, axis=1)
+
+        # Camera std of the pixel stds
+        cam_std_of_pix_std = np.ma.std(pix_std, axis=1)
+
+        # Detect outliers based on the deviation of the median distribution
+        median_deviation = pix_median - cam_median_of_pix_median[:, np.newaxis]
```

Comment:
Why is the outlier detection different to the other class?
The computation of the statistics is differente, because its a different extraction method, but why is the outlier detection different?

I think we really should split computation of statistics and deciding which pixels are "suspicious" into two separate steps.

That also makes different methods of statistics extraction and different methods of outlier detection composable

Reply:
See comment reply above.